File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

input = sys.stdin.readline
n, m, v = map(int, input().split())
adj = [[] for _ in range(n + 1)]

for i in range(m):
    u, w = map(int, input().split())
    adj[u].append(w)
    adj[w].append(u)

dfs_visited = [0] * (n + 1)
bfs_visited = [0] * (n + 1)
dfs_print = []
bfs_print = []

def dfs(v):
    if dfs_visited[v] == 0:
        dfs_print.append(v)
        dfs_visited[v] = 1
        for i in adj[v]:
            dfs(i)

dfs(v)

q = deque([v])
bfs_visited[v] = 1

while q:
    v = q.popleft()
    bfs_print.append(v)
    for i in adj[v]:
        if bfs_visited[i] == 0:
            bfs_visited[i] = 1
            q.append(i)

print(*dfs_print)
print(*bfs_print)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java

import java.util.*;
import java.io.*;
public class Main{
    static int N,M,V;
    static List<List<Integer>> adj = new ArrayList<>();
    static boolean[] visited;
    static StringBuilder sb = new StringBuilder();
    public static void main(String[] args) throws IOException{
        input();
        dfs(V);
        System.out.println(sb.toString());
        init();
        bfs(V);
        System.out.println(sb.toString());
    }
    static void input() throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());

        // 인접리스트 초기화
        for (int i=0;i<=N;i++){
            adj.add(new ArrayList<>());
        }

        // 방문배열 초기화
        visited = new boolean[N+1];

        // 인접리스트 세팅
        for (int i=0; i<M; i++){
            String[] arr = br.readLine().split(" ");
            int a = Integer.parseInt(arr[0]);
            int b = Integer.parseInt(arr[1]);
            adj.get(a).add(b);
            adj.get(b).add(a);
        }

        // 인접리스트 정렬
        for (int i=0; i<=N;i++){
            Collections.sort(adj.get(i));
        }

    }
    static void dfs(int node){
        visited[node] = true;
        sb.append(node).append(" ");
        for (int nextNode : adj.get(node)){
            if (!visited[nextNode]){
                dfs(nextNode);
            }
        }

    }
    static void bfs(int node){
        Queue<Integer> q = new LinkedList<>();
        q.offer(node);
        visited[node] = true;
        while (!q.isEmpty()){
            int currNode = q.poll();
            sb.append(currNode).append(" ");
            for (int nextNode : adj.get(currNode)){
                if (!visited[nextNode]){
                    visited[nextNode] = true;
                    q.offer(nextNode);
                }
            }
        }
    }

    static void init(){
        visited = new boolean[N+1];
        sb = new StringBuilder();
    }
}